experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl

```diff
@@ -0,0 +1,163 @@
+# bcfile_reader
+# - coordinates reading of boundary conditions from NetCDF files, as well as regridding calls and temporal interpolations from monthly to daily intervals
+
+# IO - init
+"""
+    BCDataInfo
+
+Stores information specific to each boundary condition from a file and each variable.
+"""
+struct BCFileInfo{S, V, R, D, C, O, I}
+    datafile_cgll::S
+    varname::V
+    weightfile::S
+    regrid_space::R
+    all_dates::D
+    segment_idx::Vector{Int}
+    segment_idx0::Vector{Int}
+    monthly_fields::C
+    segment_length::Vector{Int}
+    scaling_function::O # TODO: turn this into a func for scaling + offsets etc
+    interpolate_monthly::I
+end
+
+"""
+    Bcfile_info_init(datafile_rll, varname, boundary_space; interpolate_monthly = false, segment_idx0 = [Int(1)], scaling_function = false)
+
+Regrids from lat-lon grid to cgll grid, saving the output in a new file, and returns the info packaged in a single struct
+"""
+function bcfile_info_init(
+    datafile_rll,
+    varname,
+    boundary_space;
+    interpolate_monthly = false,
+    segment_idx0 = [Int(1)],
+    scaling_function = no_scalling,
+)
+
+    # regrid all times and save to file
+    weightfile, datafile_cgll, regrid_space =
+        ncreader_rll_to_cgll_from_space(datafile_rll, varname, boundary_space, outfile = varname * "_cgll.g")
+    ds = Dataset(datafile_cgll, "r")
+
+    # init time tracking info
+    current_fields =
+        interpolate_monthly ? (Fields.zeros(FT, boundary_space), Fields.zeros(FT, boundary_space)) :
+        (Fields.zeros(FT, boundary_space),)
+    segment_length = [Int(0)]
+
+    if "time" in ds
+        data_dates = Dates.DateTime.(ds["time"][:])
+    elseif "date" in ds
+        data_dates = strdate_to_datetime.(string.(ds["date"][:]))
+    else
+        @warn "No dates availabe in file $datafile_rll"
+        data_dates = nothing
+    end
+
+    return BCFileInfo(
+        datafile_cgll,
+        varname,
+        weightfile,
+        regrid_space,
+        data_dates,
+        segment_idx0 .- Int(1),
+        segment_idx0,
+        current_fields,
+        segment_length,
+        scaling_function,
+        [interpolate_monthly],
+    ) # TODO: generalize to DateTime (only dialy resol now)
+
+end
+
+no_scalling(x) = swap_space!(x, axes(mask)) # TODO: mask should not be global
```

Comment:
no_scaling?

Reply:
maybe pass in mask as an argument, then it's no longer a global?

Reply:
I actually moved this to the `bcfile_reader.jl`, where it's easy to obtain it. :)